buku

```diff
@@ -1231,146 +1265,139 @@ class BukuDb:
         if not MYHEADERS:
             gen_headers()
 
-        cond = threading.Condition()
-        cond.acquire()
-
-        def refresh(count, cond):
+        def refresh(thread_idx, cond):
             """Inner function to fetch titles and update records.
 
             Parameters
             ----------
-            count : int
-                Dummy input to adhere to convention.
+            thread_idx : int
+                Thread index/ID.
             cond : threading condition object.
             """
 
-            count = 0
+            _count = 0
 
             while True:
                 query = 'UPDATE bookmarks SET'
                 arguments = []
 
-                cond.acquire()
-                if resultset:
-                    id, url, tags, flags = resultset.pop()
-                else:
-                    cond.release()
-                    break
-                cond.release()
+                with cond:
+                    if resultset:
+                        id, url, tags, flags = resultset.pop()
+                    else:
+                        break
```

Comment:
Not actually changing any behaviour here, but using `with` does away with explicit `.release()`